### PR TITLE
Migrate Geoportal<->mp-data_manager integration into mp-layers

### PR DESCRIPTION
**Pull request by rhodges**
This pull request introduces new functionality for mapping catalog records to visualizable layers, along with improvements to the `Theme` model, integrating Layer admin forms with a paired GeoPortal catalog instance, and test coverage. It adds two new properties to the `Theme` model for retrieving direct and all descendant layers, implements a new view and endpoint for catalog-to-layer mapping, and provides comprehensive tests for these features. Additionally, it includes minor JavaScript and CSS tweaks and some cleanup in template scripts.

**New functionality for catalog mapping and Theme model:**

* Corrects broken code copied from mp-data_manager to dynamically alter Layer form options when tied to a catalog record in GeoPortal
* Added `layers` and `all_layers` properties to the `Theme` model to retrieve direct and all descendant visible layers, respectively. These are used to identify "visualizable" layers for the catalog.
* Introduced the `get_portal_catalog_map` view and corresponding endpoint `/get_portal_catalog_map`, which returns a mapping from catalog record names to layer primary keys for visible layers under top-level themes. [[1]](diffhunk://#diff-8311850ef3defe8803e74dfe1ae642e0c15e7f054a9d90750639e294e720eec2R587-R602) [[2]](diffhunk://#diff-70cfabda054792ce2008f441f4f081b7272fcf3706c132212d9f573d12afd019R21)

**Testing improvements:**

* Added thorough tests for the new `Theme.layers`, `Theme.all_layers` properties, and the `get_portal_catalog_map` view, covering edge cases and visibility rules.

**JavaScript and CSS adjustments:**

* Improved error handling and utility functions in catalog JavaScript, including a new `union` function and error logging for AJAX failures. [[1]](diffhunk://#diff-4d7982dbbc4c133095be6365117fb650492e3c39a0c2ef4cbcb73b0717768eceR7-R12) [[2]](diffhunk://#diff-4d7982dbbc4c133095be6365117fb650492e3c39a0c2ef4cbcb73b0717768eceR34-R38)
* Minor CSS tweaks to improve form field appearance in the admin interface.
* Fixed the order of JavaScript initialization in the admin layer form. [[1]](diffhunk://#diff-05d0ba07057ec562e3f034e3cffcd96c08e10013ee0e0bf7043c81ffd4e40944L25-L31) [[2]](diffhunk://#diff-05d0ba07057ec562e3f034e3cffcd96c08e10013ee0e0bf7043c81ffd4e40944L94-R93)

**Template cleanup:**

* Removed unused Knockout.js script imports from admin change form templates. [[1]](diffhunk://#diff-80cc9bf738224a626f7f506082aaaf05dc2f4f2d26587569b8bc677f6e69144eL14) [[2]](diffhunk://#diff-eae30e1721400c44777138ec980954e24ddeb54042c14c2f85f4777f0dd71a87L14)

### CODE REVIEW

File: layers/models.py [hunk 1]
```diff
@@ -455,6 +455,38 @@ def orders(self):
         content_type = ContentType.objects.get_for_model(self.__class__)
         return ChildOrder.objects.filter(object_id=self.id, content_type=content_type)
 
+    # This property gets an array of all of the Layer records that are direct children of this Theme
+    # This is particularly used by the 'get_portal_catalog_map' view to identify 'visualizable' layers
+    # in the catalog.
```
Comment (paigewilliams) [inline layers/models.py]: Love the comments. Would it be helpful to have these as doc strings for the properties so that they are visible in vscode?

File: layers/views.py [hunk 2]
```diff
@@ -584,6 +584,22 @@ def get_catalog_records(request):
 
     return JsonResponse(data)
 
+# This allows the catalog page to link to 'visualize' layers, which requires knowing the layerID that corresponds to a
```
Comment (paigewilliams) [inline layers/views.py]: this could also be a docstring

File: layers/views.py [hunk 3]
```diff
@@ -584,6 +584,22 @@ def get_catalog_records(request):
 
     return JsonResponse(data)
 
+# This allows the catalog page to link to 'visualize' layers, which requires knowing the layerID that corresponds to a 
+# given catalog record. Only layers with a catalog_name field will be included in this mapping.
+def get_portal_catalog_map(request):
+    data = {}
+    if settings.CATALOG_TECHNOLOGY == "GeoPortal2":
```
Comment (paigewilliams) [inline layers/views.py:592]: Not sure how we feel about constants in this codebase, but if `GeoPortal2` is used elsewhere it might be nice to make that a constant.

File: layers/models.py [hunk 4]
```diff
@@ -455,6 +455,38 @@ def orders(self):
...
+    # This property gets an array of all of the Layer records that are descendants of this Theme
+    # This is particularly used by the 'get_portal_catalog_map' view to identify 'visualizable' layers in the catalog.
+    @property
+    def all_layers(self):
+        content_type = ContentType.objects.get_for_model(Theme)
+        child_theme_orders = ChildOrder.objects.filter(parent_theme=self, content_type=content_type).order_by('order', 'id')
+        layers = self.layers
+        for child_order in child_theme_orders:
+            if child_order.content_object.is_visible:
+                layers = layers + child_order.content_object.all_layers
```
Comment (paigewilliams) [inline layers/models.py:484]: likely a python noob question: Why did you use `+` here and not `append`?

File: layers/models.py (hunk 1 above)
Comment (rhodges) [reply layers/models.py]: I was not aware of these features. I see there are tradeoffs, but the doc string approach looks more fun, and like an excellent step toward improving future generated documents. Thanks!

File: layers/views.py (hunk 3 above)
Comment (rhodges) [reply layers/views.py:592]: This would be an improvement, but I think will need to wait until 'next time'. This string (which I believe gets checked in MANY places) is already obsolete - we are moving to GeoPortal v3 as part of this effort, but fortunately the GeoPortal v2 logic is compatible with v3 (but not with v1). Perhaps it should be "GeoPortal2+"

File: layers/models.py (hunk 4 above)
Comment (rhodges) [reply layers/models.py:484]: Slop and bad habits from old languages. `append` would be much more readable.

Comment (rhodges) [reply layers/models.py:484]: I see I used `append` just 7 lines later - such inconsistency! I think Claude wrote the `append` line....

Comment (rhodges) [reply layers/models.py:484]: I TAKE IT BACK! On closer inspection these ARE different. In Python `append` inserts the entire list as a single entity into another list (length = len(list1) + 1), where `+` concatenates the two (length = len(list1) + len(list2)). Instead you could use `extend` here.

Comment (rhodges) [reply layers/models.py:484]:
```
>>> foo = ['a', 'b', 'c']
>>> bar = ['1', '2', '3']
>>> foo + bar
['a', 'b', 'c', '1', '2', '3']
>>> foo.append(bar)
>>> foo
['a', 'b', 'c', ['1', '2', '3']]
>>> foo = ['a', 'b', 'c']
>>> foo.extend(bar)
>>> foo
['a', 'b', 'c', '1', '2', '3']
```

File: layers/models.py (hunk 1 above)
Comment (rhodges) [reply layers/models.py]: New commit applies this.

File: layers/views.py (hunk 2 above)
Comment (rhodges) [reply layers/views.py]: New commit applies this.
